lib/streamlit/connections/snowflake_connection.py

```diff
@@ -262,27 +408,102 @@ def write_pandas(
         return (success, nchunks, nrows)
 
     def cursor(self) -> SnowflakeCursor:
-        """Return a PEP 249-compliant cursor object.
+        """Create a new cursor object from this connection.
```

Comment:
@vdonato I'm a little unclear here. This uses a command that does indeed create a new cursor, but from my tests, it only creates a new cursor on each rerun.

It's not fully cached like the connection itself (which retains the same memory ID from rerun to rerun). If I call `.cursor()` multiple times in one rerun, I get the same object returned, but on the next rerun, it's a different object. Is that strictly intentional?

Reply:
I posted an example in Slack, but if I slow it down with `time.sleep(1)`, it does in fact return a new cursor. However, without a delay between calls, it seems to return the same cursor. So I'm guessing it's an issue with the underlying library...